content/silver/dev/emacs.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Emacs Support with LSP
+weight: 250
+---
+
+{{< toc >}}
+
+
+Emacs LSP support allows for copper specific syntax highlighting through semantic tokens, lookup of declarations (preliminary), and showing errors and warnings in silver buffers. 
+
+## Prerequisites
+Above and beyond the standard Silver requirements, you need Maven (to build the silver lsp server), and of course, Emacs (26+ recommended).
+
+## Configuring Emacs prerequisites
+
+To get started, you will need to have the package [lsp-mode](https://emacs-lsp.github.io/lsp-mode/page/installation/)  installed. Note that some features of lsp-mode require additional packages, but configuring lsp-mode is a
+bit out of scope of this article. There's more information about the additional features in [here](https://emacs-lsp.github.io/lsp-mode/page/installation/#vanilla-emacs), if you're configuring with vanilla emacs.
+
+Once you have that, you'll also need to have the [silver-mode.el](https://github.com/melt-umn/silver/blob/develop/support/emacs/silver-mode/silver-mode.el)
+file loaded in some manner. This can be done by adding the path to the file to your `load-path`.
+
+``` elisp
+(add-to-list 'load-path "/path/to/silver/support/emacs/silver-mode/silver-mode.el")
+```
+
+But if you do use a package manager like [straight](https://github.com/radian-software/straight.el), I made a recipe for setting up the silver-mode as a package.
+
+```elisp
+(straight-use-package
+  `(silver-mode :type git :repo "melt-umn/silver" :files ("support/emacs/silver-mode/*.el")))
+```
+
+
+
+## Setting up LSP
+
+`LSP` has two components, the client and the server.
+
+### Setting up LSP Server 
```

Comment:
Silver language server specific stuff that isn't specific to the emacs plugin should probably be a separate page that is linked here.  Which I was intending to write at some point, but the steps for building the language server are a good start.

Reply:
I can move this to another page